src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue53/DumpWithoutCommentsTest.kt

```diff
@@ -0,0 +1,54 @@
+package it.krzeminski.snakeyaml.engine.kmp.issues.issue53
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.StringStreamDataWriter
+import it.krzeminski.snakeyaml.engine.kmp.api.lowlevel.Serialize
+import it.krzeminski.snakeyaml.engine.kmp.composer.Composer
+import it.krzeminski.snakeyaml.engine.kmp.emitter.Emitter
+import it.krzeminski.snakeyaml.engine.kmp.events.Event
+import it.krzeminski.snakeyaml.engine.kmp.nodes.Node
+import it.krzeminski.snakeyaml.engine.kmp.parser.ParserImpl
+import it.krzeminski.snakeyaml.engine.kmp.scanner.StreamReader
+import it.krzeminski.snakeyaml.engine.kmp.serializer.Serializer
+
+/**
+ * Issue 53.
+ */
+class DumpWithoutCommentsTest : FunSpec({
+    test("dump map with comments") {
+        val yaml = "a: 1 # A\nb: 2 # B\n"
+        val dumpSettings = DumpSettings.builder().setDumpComments(false).build()
+
+        val emitter = Emitter(
+            dumpSettings,
+            StringStreamDataWriter(),
+        )
+        val serializer = Serializer(dumpSettings, emitter)
+
+        serializer.emitStreamStart()
+        serializer.serializeDocument(createNodeWithComments(yaml))
+        serializer.emitStreamEnd()
```

Comment:
issue: looks like this test doesn't have any assertions. We should add verification that comments were dumped (really strange that this test was committed to the original snakeyaml repository...)

Reply:
Even a test without assertions implicitly asserts on not throwing an exception - maybe that was the intent? Let me create an issue to follow up on this, because I'd like this change to get merged sooner. Thanks for catching this!

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/issues/317